pkg/curatedpackages/packageclient.go

```diff
@@ -46,3 +59,74 @@ func convertBundleVersionToPackageVersion(bundleVersions []packagesv1.SourceVers
 	}
 	return versions
 }
+
+func (pc *PackageClient) GeneratePackages() ([]packagesv1.Package, error) {
+	packageMap := pc.getPackageNameToPackage()
+	var packages []packagesv1.Package
+	for _, p := range pc.packages {
+		bundlePackage := packageMap[strings.ToLower(p)]
+		if bundlePackage.Name == "" {
+			fmt.Println(fmt.Errorf("unknown package %s", p).Error())
+			continue
+		}
```

Comment:
```suggestion
		bundlePackage, found := packageMap[strings.ToLower(p)]
		if !found {
			fmt.Fprintf(os.Stderr, "unknown package %q\n", p)
			continue
		}
```

This is a safer, and more idiomatic way to check for the presence of a value within a map. It's safer because if the element type of the map were to be a pointer, then the value would be nil, and you'd be dereferencing a nil pointer, which is of course a run-time error.

Also, `fmt.Println` doesn't take "%s" holders.

There's no need to generate an error, just to call its `Error()` function.

Lastly, error output should generally go to [`os.Stderr`](https://pkg.go.dev/os#pkg-variables).

PS- We should consider if we want to continue in the face of an unknown package. If said package were a dependency for something else, and the user didn't notice the error output, we could possibly do something destructive.

Reply:
I did consider whether to continue or not, I decided to go with continuing and the reason is because the result here would be creating a package configuration and there is no expectation of dependency between the packages.

Reply:
How can we know that?